iib/workers/tasks/build.py

```diff
@@ -411,10 +428,7 @@ def _push_image(request_id, arch):
             'fixing it with skopeo.',
             destination,
         )
-        run_cmd(
-            ['skopeo', 'copy', '--format', 'v2s2', destination, destination],
-            exc_msg=f'Failed to fix the manifest schema version on {destination}',
-        )
+        _fix_schema_version(destination)
```

Comment:
Could you please change the `_push_image` test so that it mocks out `_fix_schema_version` and create a new test for `_fix_schema_version`? This makes it so that the test for `_push_image`  is not reliant on the way `_fix_schema_version` and it makes the tests less brittle and more independent.